htcanalyze/host_nodes.py

```diff
@@ -0,0 +1,99 @@
+"""Class to store host nodes."""
+import socket
+import logging
+from datetime import timedelta
+
+
+class NodeCache:
+    """Cache to save reverse DNS lookups."""
+
+    def __init__(self):
+        self.rdns_cache = dict()
+
+    def gethostbyaddrcached(self, ip):
+        """
+        Get the hostname by address, with an in-memory cache.
+
+        This prevents excessive queries to DNS servers.
+
+        :param ip: ip represented by a string
+        :return: resolved domain name, else give back the IP
+        """
+        try:
+            # try our cache first
+            return self.rdns_cache[ip]
+        except KeyError:
+            # do the lookup
+            try:
+                rdns = socket.gethostbyaddr(ip)
```

Comment:
Shortened:
```
self.rdns_cache[ip] = socket.gethostbyaddr(ip)[0]
logging.debug(f"rDNS lookup successful {ip} resolved as {self.rdns_cache}")
return self.rdns_cache
```

Change `socket.get_host_by_address` (or `socket.get_host_by_ip`?)

Why set both `self.rdns_cache` and return the same value? Are both necessary, because usually you only do one of both.

Reply:
Haha that makes me laugh 😆, I will not touch the socket library methods, 
but yeah maybe there could be some improvement. 
Alex has made this code segment, so maybe his thoughts have their reasons.